WaveManager/src/main/java/com/github/sef24sp4/wavemanager/WaveManager.java

```diff
@@ -0,0 +1,201 @@
+package com.github.sef24sp4.wavemanager;
+
+import com.github.sef24sp4.common.enemy.CommonEnemy;
+import com.github.sef24sp4.common.enemy.EnemyRole;
+import com.github.sef24sp4.common.enemy.EnemySPI;
+import com.github.sef24sp4.common.entities.IEntity;
+import com.github.sef24sp4.common.interfaces.IGameSettings;
+import com.github.sef24sp4.common.vector.Coordinates;
+
+import java.util.*;
+
+public class WaveManager implements IWaveManager {
+
+	private Collection<IEntity> enemyComposition = new HashSet<>();
+
+	/**
+	 * used instead of a map.
+	 */
+	private final IGameSettings gameSettings;
+
+	private int waveNumber;
+
+	private long waveDifficulty;
+
+	private final int timeUntilNextWaveInSeconds;
+
+	private long timeOfLastCheck;
+
+	private WaveStatus waveStatus = WaveStatus.ONGOING;
```

Comment:
Is this the most appropriate status? If so, then maybe put in a comment explaining why it is more appropriate than `WaveStatus.READY`.

Reply:
here i am again assuming this have nothing to do with line 26 - 28, and its because of the switch statement in the control system. the control system calls the nextwave method and to do so it needs to be in that state, it is the state where it checks wether or not it can be ready.